### PR TITLE
Fix refund date constraints to prevent future dates

### PR DESCRIPTION
**Pull request by layoutd**
## Summary
- Prevents refund dates from being set in the future
- Ensures second refunds always occur after first refunds
- Handles edge cases where time windows are insufficient

## Related
Builds on: #182

## Changes
Modified `calculate_refund_date()` in `Order.php`:
- Added `time()` checks to cap refund dates at current time
- For first refunds: ensures date is between order completion and now (capped at 60 days)
- For second refunds: ensures date is between first refund and now (capped at 30 days), with minimum 1-second gap
- Added fallback logic for edge cases:
  - First refund: if completion is in future, uses current time; if no time window, uses completion timestamp
  - Second refund: if no time window, uses first refund timestamp + 1 hour

## Technical Details
**Before:**
- First refund: `completion_date + random(0-60 days)` → could be in future
- Second refund: `first_refund_date + random(0-30 days)` → could be in future

**After:**
- First refund: `random(completion_date, min(completion_date + 60 days, now))` (with edge case fallbacks)
- Second refund: `random(first_refund_date + 1 second, min(first_refund_date + 30 days, now))` (or `first_refund_date + 1 hour` if no time window)

## Test Plan
- Generate orders from this month and verify refunds date/time doesn't exceed current time
    `wp wc generate orders 10 --date-start=2025-10-01 --status=completed --refund-ratio=1.0`
- Generate orders with --date-start in the past and verify refund dates are realistic (all in Q1 2025)
    `wp wc generate orders 10 --date-start=2025-01-01 --date-end=2025-01-31 --status=completed --refund-ratio=1.0`
- Verify second refunds always have timestamps after first refunds (note they display in reverse order in the UI)

### CODE REVIEW

File: includes/Generator/Order.php [hunk 1]
```diff
@@ -771,23 +771,44 @@ protected static function calculate_refund_totals( $line_items ) {
...
+			// Second refund: must be after first refund but before current time
+			$base_timestamp = strtotime( $previous_refund->get_date_created()->date( 'Y-m-d H:i:s' ) );
+			$max_timestamp = min( $base_timestamp + ( self::SECOND_REFUND_MAX_DAYS * DAY_IN_SECONDS ), $now );
+
+			// Ensure second refund is always after first refund
+			if ( $max_timestamp <= $base_timestamp ) {
+				// If there's no time window, use base timestamp + 1 hour (or now if that's in future)
+				$refund_timestamp = min( $base_timestamp + HOUR_IN_SECONDS, $now );
+			} else {
+				$refund_timestamp = wp_rand( $base_timestamp + 1, max( $base_timestamp + 1, $max_timestamp ) );
```
Comment (chihsuan) [inline includes/Generator/Order.php]: If I understand correctly, the `max()` call is redundant since we already know `$max_timestamp > $base_timestamp`?

File: includes/Generator/Order.php [hunk 2]
```diff
@@ -771,23 +771,44 @@ protected static function calculate_refund_totals( $line_items ) {
...
-			$base_date = $previous_refund->get_date_created();
-			$max_days = self::SECOND_REFUND_MAX_DAYS;
+			// Second refund: must be after first refund but before current time
+			$base_timestamp = strtotime( $previous_refund->get_date_created()->date( 'Y-m-d H:i:s' ) );
+			$max_timestamp = min( $base_timestamp + ( self::SECOND_REFUND_MAX_DAYS * DAY_IN_SECONDS ), $now );
+
+			// Ensure second refund is always after first refund
+			if ( $max_timestamp <= $base_timestamp ) {
+				// If there's no time window, use base timestamp + 1 hour (or now if that's in future)
+				$refund_timestamp = min( $base_timestamp + HOUR_IN_SECONDS, $now );
```
Comment (chihsuan) [inline includes/Generator/Order.php]: Does this mean the second refund could occur before the first if the first is in the future?

**Review by chihsuan — COMMENTED** [review]
Appreciate the fix, @layoutd! I've verified it's working and left a few comments.

It occurs to me that there is a timezone issue, since we're using UTC to calculate the date. In the past, I encountered a case where orders appeared to be created in the future because my site’s time zone wasn’t UTC. However, I think we can leave this for later.

File: includes/Generator/Order.php (hunk 2 above)
Comment (layoutd) [reply includes/Generator/Order.php]: Yeah, I guess if the first refund is in the future, that could happen. Removed the limit in 92130a8fe03eabd37ccb5f0ef4ad12618533106f

File: includes/Generator/Order.php (hunk 1 above)
Comment (layoutd) [reply includes/Generator/Order.php]: 🔍 Nice catch 2a68624501534c45f4753f4489233bda087d82b7

**Review by chihsuan — APPROVED** [review]
LGTM! 🚢
